src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlFormatGeneratorStatics.cs

```diff
@@ -596,6 +605,9 @@ internal static MethodInfo CheckEndOfArrayMethod
         private static MethodInfo? s_getArrayLengthMethod;
         internal static MethodInfo GetArrayLengthMethod
         {
+            [DynamicDependency(DynamicallyAccessedMemberTypes.PublicProperties, typeof(Array))]
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2075:RequiresUnreferencedCode",
```

Comment:
What do you think about just removing `Globals.TypeOfArray` and just using `typeof(Array)` inline here? Then the trimmer wouldn't warn at all.

Reply:
We could do that, but it would of course mean that we wouldn't be using the cached typeof call any longer and instead calling typeof everytime. With this PR I tried to not change behavior of the library as much as possible, given most of its use is unsafe anyway. I think both what you suggested and what I have addresses the warning correctly, but mine doesn't change behavior, so unless you feel strongly about it, I would keep it as is.

Reply:
We should be limiting DynamicDependency + UnconditionalSuppressMessage as much as possible. It is basically a hack when you need to do it. If there is a way to write the code without it, we should take it.

Reply:
> that we wouldn't be using the cached typeof call any longer and instead calling typeof everytime.

Note that this code is only called 1 time anyway since the MethodInfo is cached into `s_getArrayLengthMethod`.

Reply:
oh you are correct, that is a fair point. I'll ping Matt to let him know and  make the change.

Reply:
>We should be limiting DynamicDependency + UnconditionalSuppressMessage as much as possible. It is basically a hack when you need to do it.

BTW, is this what we plan to tell library devs too? I ask because at least in my mind it is more of a "workaround" or a "help to the linker" than an actual hack.

Reply:
Yes. If there is a way to write your code such that you don't need to suppress messages, you should write it that way. `DynamicDependency` isn't a sustainable approach, and should be used as little as possible. Sometimes it is the only way to make things work, but here it isn't.